src/main/java/net/dries007/tfc/ForgeEventHandler.java

```diff
@@ -627,6 +627,15 @@ else if (block == TFCBlocks.BLOOMERY.get() && !state.getValue(BloomeryBlock.LIT)
                 event.setCanceled(true);
             }
         }
+        else if (block == TFCBlocks.POWDERKEG.get() && state.getValue(PowderkegBlock.SEALED))
+        {
+            final BlockEntity entity = level.getBlockEntity(pos);
+            if (entity instanceof PowderkegBlockEntity)
+            {
+                ((PowderkegBlockEntity) entity).setLit(true);
+                event.setCanceled(true);
+            }
```

Comment:
```suggestion
            level.getBlockEntity(pos, TFCBlockEntities.POWDERKEG).ifPresent(entity -> {
                entity.setLit(true);
                event.setCanceled(true);
            });
```
- Cleans up the `instanceof` cast
- Specifically queries for the block entity type rather than relying on an `instanceof`